src/components/Trade/hooks/useSwapper/useSwapper.ts

```diff
@@ -81,13 +79,40 @@ export const useSwapper = () => {
     manager.addSwapper(SwapperType.Zrx, new ZrxSwapper({ web3, adapterManager }))
     return manager
   })
-  const [bestSwapperType, setBestSwapperType] = useState(SwapperType.Zrx)
+
+  const [bestSwapperType] = useState(SwapperType.Zrx)
   const [debounceObj, setDebounceObj] = useState<{ cancel: () => void }>()
 
+  const filterAssetsByIds = (assets: Asset[], assetIds: string[]) => {
+    const assetIdMap = Object.fromEntries(assetIds.map(assetId => [assetId, true]))
+    return assets.filter(asset => assetIdMap[asset.caip19])
+  }
```

Comment:
this can be simplified to

```
const assets = useSelector(selectAssets)
const filteredAssets: Asset[] = assetIds.map(assetId => assets[assetId])
// also wrap the above in a useCallback, i won't get the syntax right in a comment tho
```

Reply:
https://github.com/shapeshift/web/issues/1628